### PR TITLE
Exosuit Expansion: Return of the Ripley

### PR DESCRIPTION
**Pull request by Mechoid**
Background:
- Exosuit dynX calls made largely defunct, however still vestigially present for fear of anomalous properties.
- Exosuit equipment can now process alongside each-other when handling things such as armor.

Fixes:
- Shocker now retaliates properly against melee.

Tweaks:
- Ignition effects from weapons changed from the old fire-system to the modifier fire system.
- Incendiary exo weapons tweaked to conform with fire change.

Additions:
- Hardpoint Actuator equipment for Exosuits, allowing them (And most effectively, Ripleys) to swap components on the fly, after a short delay.
- Exo Inflatables Deployer, allowing exosuits to pick up and deploy inflatables.
- Exo screwdriver, wirecutters, and welding laser, to help round-out the tool-based equipment types.
- Fire modifiers based on stacks added, to be used by the damage system tweaked above.

### CODE REVIEW

File: code/modules/mob/_modifiers/fire.dm [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Ignition, but confined to the modifier system.
+// This makes it more predictable and thus, easier to balance.
+/datum/modifier/fire
+	name = "on fire"
+	desc = "You are on fire! You will be harmed until the fire goes out or you extinguish it with water."
+	mob_overlay_state = "on_fire"
+
+	on_created_text = "<span class='danger'>You combust into flames!</span>"
```
Comment (Atermonera) [inline code/modules/mob/_modifiers/fire.dm]: dictionary
`You [are consumed by fire] into flames!`
Burst, perhaps?

File: code/modules/mob/living/damage_procs.dm [hunk 2]
```diff
@@ -109,5 +109,9 @@
 	if(drowsy)		apply_effect(drowsy, DROWSY, blocked)
 	if(agony)		apply_effect(agony, AGONY, blocked)
 	if(flammable)	adjust_fire_stacks(flammable)
-	if(ignite)		IgniteMob()
+	if(ignite)
+		if(ignite >= 3)
+			add_modifier(/datum/modifier/fire/stack_managed/intense, 60 SECONDS)
```
Comment (Atermonera) [inline code/modules/mob/living/damage_procs.dm]: If ignite = 2, you will have regular fire for 90s, but if ignite = 15, you'll only burn for 60s, intentional?

Comment (Mechoid) [reply code/modules/mob/living/damage_procs.dm]: Intentional, as 60s of intense fire is 600 damage, while 90 seconds of standard is 450 damage.

File: code/game/mecha/equipment/mecha_equipment_dynamicprocs.dm [hunk 3]
```diff
@@ -4,18 +4,15 @@
...
 // Used for impacting (thrown) objects, and damage value.
 /obj/item/mecha_parts/mecha_equipment/proc/handle_ranged_contact(var/obj/A, var/inc_damage = 0)
-	if(!inc_damage)
-		return 0
-
-	return inc_damage
+	return max(0, inc_damage)
 
 // Used for melee strikes with an object, and a mob, and damage value.
 /obj/item/mecha_parts/mecha_equipment/proc/handle_melee_contact(var/obj/item/weapon/W, var/mob/living/user, var/inc_damage = null)
```
Comment (Atermonera) [inline code/game/mecha/equipment/mecha_equipment_dynamicprocs.dm]: inc_damage should default to 0, for consistency

File: code/game/mecha/equipment/mecha_equipment_dynamicprocs.dm [hunk 4]
```diff
@@ -4,18 +4,15 @@
...
-	return inc_damage
+	return max(0, inc_damage)
 
 // Used for projectile impacts from bullet_act.
 /obj/item/mecha_parts/mecha_equipment/proc/handle_projectile_contact(var/obj/item/projectile/Proj, var/inc_damage)
```
Comment (Atermonera) [inline code/game/mecha/equipment/mecha_equipment_dynamicprocs.dm]: inc_damage should default to 0, for consistency

**Pull request merged**